.github/workflows/haskell.yml

```diff
@@ -0,0 +1,53 @@
+name: Binaries
+
+defaults:
+  run:
+    shell: bash
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        ghc: ["8.10.7"]
+        os: [ubuntu-latest]
+        aeson: ["1.5.6.0", "2.0.3.0"]
```

Comment:
Add a comment explaining what this is for, namely testing that we do actually have aeson 1/2 compatibility.

Reply:
Explanations added.